### ISSUE TITLE
liblrs can't be used on macos

### ISSUE THREAD

**Issue by bloussou** [OpenRailAssociation/liblrs#83]
```
Installing liblrs-python (0.1.7): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  Running `maturin pep517 build-wheel -i /private/var/folders/rn/_mpvgskn6gvff23095bsvg640000gq/T/tmp8q576_hu/.venv/bin/python --compatibility off`
  🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.8
  🐍 Not using a specific python interpreter
  📡 Using build options features from pyproject.toml
  💻 Using `MACOSX_DEPLOYMENT_TARGET=11.0` for aarch64-apple-darwin by default
  error: package `clap v4.5.4` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.73.0
  Either upgrade to rustc 1.74 or newer, or use
  cargo update -p clap@4.5.4 --precise ver
  where `ver` is the latest version of `clap` supporting rustc 1.73.0
  💥 maturin failed
    Caused by: Failed to build a native library through cargo
    Caused by: Cargo build finished with "exit status: 101": `env -u CARGO MACOSX_DEPLOYMENT_TARGET="11.0" PYO3_ENVIRONMENT_SIGNATURE="cpython-3.11-64bit" PYO3_PYTHON="/private/var/folders/rn/_mpvgskn6gvff23095bsvg640000gq/T/tmp8q576_hu/.venv/bin/python" PYTHON_SYS_EXECUTABLE="/private/var/folders/rn/_mpvgskn6gvff23095bsvg640000gq/T/tmp8q576_hu/.venv/bin/python" "cargo" "rustc" "--features" "pyo3/extension-module" "--message-format" "json-render-diagnostics" "--manifest-path" "/private/var/folders/rn/_mpvgskn6gvff23095bsvg640000gq/T/tmprf3knn4b/liblrs_python-0.1.7/python/Cargo.toml" "--release" "--lib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/liblrs_python.abi3.so" "-C" "strip=symbols"`
  Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/private/var/folders/rn/_mpvgskn6gvff23095bsvg640000gq/T/tmp8q576_hu/.venv/bin/python', '--compatibility', 'off'] returned non-zero exit status 1
```

**Comment by Tristramg** [comment]
Arg, this is anoying. Did it happen when installing the depedency in a project or did you try to build it directly on your computer?

**Comment by Tristramg** [comment]
Should be closed by #84

**Comment by Tristramg** [comment]
We tested on a mac, it seems to work fine
